src/app/content/highlights/components/ColorPicker.tsx

```diff
@@ -62,33 +63,88 @@ const ColorButton = styled(({className, size, style, ...props}: ColorButtonProps
+  if (key === 'Home') {
+    return 0;
+  }
+  if (key === 'End') {
+    return itemCount - 1;
+  }
+  if (key === 'ArrowLeft') {
+    return (idx + itemCount - 1) % itemCount;
+  }
+  // ArrowRight
+  return (idx + 1) % itemCount;
+}
+
 // tslint:disable-next-line:variable-name
 const ColorPicker = ({className, ...props}: Props) => {
+  const ref = React.useRef<HTMLDivElement>();
+  // https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/radiogroup_role#keyboard_interactions
+  const handleKeyNavigation = React.useCallback(
+    (event: React.KeyboardEvent) => {
+      if (!['ArrowLeft', 'ArrowRight', 'Home', 'End'].includes(event.key)) {
+        return;
+      }
+      const assertedRef = ref as React.MutableRefObject<HTMLDivElement>;
+      const activeElement = document?.activeElement as HTMLInputElement;
+      const inputs = Array.from(assertedRef.current.querySelectorAll('input'));
+      const idx = inputs.indexOf(activeElement);
+      const destIdx = nextIdx(idx, inputs.length, event.key as NavKeys);
+      const el = inputs[destIdx];
+
+      event.preventDefault();
+      el.focus();
+      el.click();
+    },
+    []
+  );
+  const initialFocus = React.useCallback(
+    (event: React.FocusEvent) => {
+      if (!ref.current || event.currentTarget !== event.target) {
+        return;
+      }
+      // There will always be one checked
+      (ref.current.querySelector<HTMLInputElement>('input[checked]') as HTMLInputElement).focus();
+    },
+    []
+  );
 
-  // its important that this be click focusable, because when clicking
-  // labels in chrome the focus is first moved to a background element
-  // before the input, and that might cause weird behavior in parent
-  // elements.
-  return <div className={className} tabIndex={-1}>
-    {highlightStyles.map((style) => <ColorButton key={style.label}
-      name={style.label}
-      checked={props.multiple ? props.selected.includes(style.label) : props.color === style.label}
-      style={style}
-      size={props.size}
-      onChange={() => props.multiple
-        ? props.selected.includes(style.label)
-          ? props.onChange(props.selected.filter(not(match(style.label))))
-          : props.onChange([...props.selected, style.label])
-        : props.color === style.label
-          ? props.onRemove ? props.onRemove() : null
-          : props.onChange(style.label)}
-    />)}
-  </div>;
+  return (
+    <div
+      className={className}
+      tabIndex={0}
+      role='radiogroup'
+      aria-label='colors'
+      ref={ref as unknown as LegacyRef<HTMLElement>}
+      onKeyDown={handleKeyNavigation}
+      onFocus={initialFocus}
+    >
+      {highlightStyles.map((style) => <ColorButton key={style.label}
+        name={style.label}
+        checked={props.multiple ? props.selected.includes(style.label) : props.color === style.label}
+        style={style}
+        size={props.size}
+        tabindex={-1}
```

Comment:
React wants this to be `tabIndex` which I never remember until I randomly notice it yelling at me in the console